.github/workflows/test-and-build.yml

```diff
@@ -51,3 +51,36 @@ jobs:
 
             - name: Run unit tests
               run: pnpm test
+
+            # AI Agent Tests to run only after file changes
+            - name: Check for AI files changes
+              uses: dorny/paths-filter@v3
+              id: ai_changes
+              with:
+                  token: ${{ github.token }}
+                  filters: |
+                    ai:
+                      - 'packages/backend/src/ee/services/ai/**/*'
+                      - 'packages/common/**/*'
+
+            - uses: 8BitJonny/gh-get-current-pr@3.0.0
+              id: PR
+              if: github.event_name == 'pull_request'
+
+            - name: Check for AI test override if the PR description contains "test-ai"
+              id: ai_override
+              run: |
+                  if [ "${{ github.event_name }}" = "pull_request" ]; then
+                      echo "AI override: ${{contains(steps.PR.outputs.pr_body, 'test-ai')}}"
+                      echo "should_run=${{contains(steps.PR.outputs.pr_body, 'test-ai')}}" >> "$GITHUB_OUTPUT"
+                  else
+                      echo "should_run=false" >> "$GITHUB_OUTPUT"
+                  fi
+
+            - name: Run AI Agent Tests
+              if: steps.ai_changes.outputs.ai == 'true' || steps.ai_override.outputs.should_run == 'true'
+              run: pnpm --filter backend test -- --testPathPattern="src/ee/services/ai/agents/tests/agent.test.ts"
```

Comment:
would be nice to have it:
- under pnpm `test:ai`
- include everything under `services/ai`
- exclude services/ai from `pnpm test`